var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -186,8 +186,22 @@ def mpi_dependent_options(self):
             compiler_opts = [
                 '--with-cc=%s' % self.spec['mpi'].mpicc,
                 '--with-cxx=%s' % self.spec['mpi'].mpicxx,
-                '--with-fc=%s' % self.spec['mpi'].mpifc
+                '--with-fc=%s' % self.spec['mpi'].mpifc,
             ]
+            if self.spec.satisfies('%intel'):
+                # "Inspired" by what is done in the elemental package
+                # ifort doesn't automatically link all run-time libraries
+                ifort = env['SPACK_F77']
+                intel_bin = os.path.dirname(ifort)
+                intel_lib_root = os.path.join(
```

Comment:
since https://github.com/spack/spack/pull/7469 I think there is a better way of doing this, is this correct @mgsternberg ?

Reply:
I'm all for a cleaner solution but I couldn't find an obvious one from the documentation linked from (the last comments) of that pull request. Maybe you have an example or pointers so any other relevant documentation? (I'll also wait for @mgsternberg feedback.)

Reply:
what we do in `trilinos` to get `libgfortran` is
```
                libgfortran = os.path.dirname(os.popen(
                    '%s --print-file-name libgfortran.a' %
                    join_path(mpi_bin, 'mpif90')).read())
```
does `ifort` has something similar? It's not much cleaner, but at least should not use version **unspecific** `lib` folder...

Reply:
As far as I undestand the `--print-file-name` option is a gnu specific one, I've looked for something similar for ifort but found nothing.
And because of this I did some more digging and a ran a few tests and it seems a much simpler `FC_LINKER_FLAGS` is sufficient to make everything work. See my next commit.